### PR TITLE
Communications services TC/TM update + test scripts

### PR DESCRIPTION
**Pull request by ArashYzd**

### CODE REVIEW

File: src/groundStation/commandParser.py [hunk 1]
```diff
@@ -152,12 +152,10 @@ def __argCheck(self, args, inoutInfo, subservice=None):
 
     def __lexer(self, input):
         tokenList = []
-<<<<<<< HEAD:src/commandParser.py
+        # '([a-zA-Z_-]+|[\(\)]|[0-9_.-]*)' #([^#]*)#|
         tmp = re.split('([-.]+|[0-9.]+|[a-zA-Z0-9_-]+|[\(\)])', input)
```
Comment (ArashYzd) [inline src/groundStation/commandParser.py:156]: Ok, we have had conversations about conflicts this regex create. I have tried a lot of different regexs (e.g., adding /S), and even though they're supposed to work, for some reason they don't parse and show errors. So, I'm keeping this one for now.

Comment (ArashYzd) [reply src/groundStation/commandParser.py:156]: A set of test cases I need to be considered as a single group: ab12x bc 4vc23 31.85 ---.--. -2 hello
And other regexes that are supposed to work but they don't:
([a-zA-Z_-]\S+|[\(\)]|[0-9_.-]\S*)
([-.]\S+|[0-9.]\S+|[a-zA-Z0-9_-]+|[\(\)])
([a-zA-Z0-9_.-]+|[\(\)]|[0-9_.-]*)

File: src/groundstation.py [hunk 2]
```diff
@@ -36,8 +36,8 @@
 import re
 if __name__ == '__main__':
     # We're running this file directly, not as a module.
-    from commandParser import CommandParser
-    from system import SystemValues
+    from groundStation.commandParser import CommandParser
```
Comment (ArashYzd) [inline src/groundstation.py:39]: How was your code working?

File: src/test_sband.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
+'''
```
Comment (ArashYzd) [inline src/test_sband.py:1]: I was thinking of a separate directory for test scripts. There was a lot of conflicts in the import commands so I gave it up. But I think it's needed. What are your thoughts?

**Review by ArashYzd — COMMENTED** [review]
I forgot to pull the new commits first...my bad. I'll resolve the conflicts once I have your feedback.

File: src/groundStation/commandParser.py [hunk 4]
```diff
@@ -152,12 +152,10 @@ def __argCheck(self, args, inoutInfo, subservice=None):
...
-=======
-        tmp = re.split('([a-zA-Z_-]+|[\(\)]|[0-9_.-]*)', input)
->>>>>>> upstream/master:src/groundStation/commandParser.py
-        [tokenList.append(x.upper()) for x in tmp if not str(x).strip() == '']
+        [tokenList.append(x.upper()) for x in tmp if not (
+            str(x).strip() == '' or str(x).strip() == ',')]  # to accept ',' as delimiter
```
Comment (ArashYzd) [inline src/groundStation/commandParser.py:158]: My intention for this was to become able to use space in the strings and use a comma instead to separate the input arguments. It was an unsuccessful attempt but will be needed if we really consider changing the morse beacon or the beacon message from GS.

Comment (ArashYzd) [reply src/groundStation/commandParser.py:158]: Example: set_beacon_msg(I am Ex-Alta2) and put the "I am Ex-Alta2" into the packet as a single string.

**Comment by ArashYzd** [comment]
@arrooney I definitely need to update this PR (probably easier to make a new one). But before doing so, would you take a look at my comments and share your thoughts, please? Thanks.

File: src/groundstation.py [hunk 5]
```diff
@@ -66,7 +66,7 @@ def __zmq__(self, addr):
         libcsp.rtable_load('0/0 ZMQHUB')
 
     def __uart__(self):
-        libcsp.kiss_init('/dev/ttyUSB15', 9600, 512, 'uart')
+        libcsp.kiss_init('/dev/ttyUSB0', 9600, 512, 'uart')
```
Comment (arrooney) [inline src/groundstation.py:69]: This is passed as an argument in the updated version, there should be no src/groundstation.py, you should pull from master, perhaps with the `--rebase` option

File: src/test_sband.py (hunk 3 above)
Comment (arrooney) [reply src/test_sband.py:1]: yeah I agree, but we should maybe have a common test library so we don't repeat the `sendAndExpect` method for instance

File: src/test_sband.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    return response == expect
+
+
+def testAllCommandsToOBC():
+    current = int(time.time())
+    sendAndExpect('obc.communication.s_set_control(0 0)', {'err': 0})
+    sendAndExpect('obc.communication.s_get_control',
+                  {'err': 0, 'status': 0, 'mode': 0})
+    sendAndExpect('obc.communication.s_get_full_status', {'err': 0, 'PWRGD': 1, 'TXL': 1, 'Transmit Ready': 1, 'Buffer Count': 0, 'Buffer Underrun': 1, 'Buffer Overrun': 2, 'Output Power': np.float32(26), 'Power Amplifier Temperature': np.float32(
+        27.3), 'Top Temperature': np.float32(-2.8), 'Bottom Temperature': np.float32(11.7), 'Battery Current': np.float32(95), 'Battery Voltage': np.float32(7.2), 'Power Amplifier Current': np.float32(0.48), 'Power Amplifier Voltage': np.float32(5.1), 'Firmware Version': np.float32(7.14)})
```
Comment (arrooney) [inline src/test_sband.py:45]: This part looks good.

File: src/test_uhf.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+opts = groundStation.options()
+gs = groundStation.groundStation(opts.getOptions())
+
+
+def sendAndExpect(send, expect):
+    server, port, toSend = gs.getInput(inVal=send)
+    response = gs.transaction(server, port, toSend)
+    testpassed = 'Pass' if response == expect else 'Fail'
+    print(' - TEST CASE ' + testpassed + ' -\n\tSent: ' + send +
+          '\n\tRecieved: ' + str(response) +
```
Comment (arrooney) [inline src/test_uhf.py:34]: This should be restructured so that this is not repeated.

**Review by arrooney — CHANGES_REQUESTED** [review]
Sorry for not getting to this earlier. The tests look good, but there are clearly some conflicts with the fact that the grounstation.py has been moved. Also I agree with making a test directory, with maybe also a /test/util/ directory where the common code is stored. But perhaps we can merge these tests (After resolving the groundstation.py issues) and make a new ticket for refactoring this test directory, since we'll probably make a new one for EPS soon-sh

**Pull request merged**